connection.go

```diff
@@ -205,13 +218,21 @@ func ConfigHandler(clientType string, config *sarama.Config) ClientConfig {
 	return conf
 }
 
-func Init(token string, config interface{}, options ...Option) {
+func Init(token string, config interface{}, options ...Option) *sarama.Config {
+
+	sconfig := config.(*sarama.Config)
+	newConfig := *sconfig
+
+	if Clients == nil {
+		Clients = make(map[int]*Client)
+	}
+
 	opts := GetDefaultOptions()
 	for _, opt := range options {
 		if opt != nil {
 			if err := opt(&opts); err != nil {
 				fmt.Printf("superstream: error initializing superstream: Wrong option: %s", err.Error())
-				return
+				return nil
```

Comment:
in case of error return the old config to not cause issues to the user, make sure to make it in all places